File: flutter/lib/models/employee.dart
class Employee {
  final int id;
  final int ownerUserId;
  final String username;
  final String? name;
  final String? phone;
  final String? role;
  final bool canManageInventory;
  final bool canManageOrders;
  final bool canChat;
  final bool canChangeStatus;
  final bool canRate;
  final bool statusActive;
  final String? profilePicUrl;
  final DateTime createdAt;
  final DateTime updatedAt;

  Employee({
    required this.id,
    required this.ownerUserId,
    required this.username,
    this.name,
    this.phone,
    this.role,
    required this.canManageInventory,
    required this.canManageOrders,
    required this.canChat,
    required this.canChangeStatus,
    required this.canRate,
    required this.statusActive,
    this.profilePicUrl,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Employee.fromJson(Map<String, dynamic> json) {
    DateTime parseDate(String key) {
      final value = json[key];
      if (value is String) {
        final parsed = DateTime.tryParse(value);
        if (parsed != null) return parsed;
      }
      return DateTime.now();
    }

    return Employee(
      id: json['id'] as int,
      ownerUserId: json['owner_user_id'] as int,
      username: json['username'] as String,
      name: json['name'] as String?,
      phone: json['phone'] as String?,
      role: json['role'] as String?,
      canManageInventory: json['can_manage_inventory'] as bool? ?? false,
      canManageOrders: json['can_manage_orders'] as bool? ?? false,
      canChat: json['can_chat'] as bool? ?? false,
      canChangeStatus: json['can_change_status'] as bool? ?? false,
      canRate: json['can_rate'] as bool? ?? false,
      statusActive: json['status_active'] as bool? ?? false,
      profilePicUrl: json['profile_pic_url'] as String?,
      createdAt: parseDate('created_at'),
      updatedAt: parseDate('updated_at'),
    );
  }
}

File: golang/models/employee.go
package models

import (
	"time"

	"gorm.io/gorm"
)

// Employee accounts are owned by a supplier or store user.
type Employee struct {
	ID                 uint           `gorm:"primaryKey" json:"id"`
	OwnerUserID        uint           `gorm:"not null;index:idx_employee_owner_username,priority:1" json:"owner_user_id"`
	Username           string         `gorm:"not null;index:idx_employee_owner_username,priority:2" json:"username"`
	Password           string         `gorm:"not null" json:"-"`
	Name               string         `json:"name"`
	Phone              string         `json:"phone"`
	Role               string         `gorm:"type:varchar(50)" json:"role"`
	CanManageInventory bool           `gorm:"default:true" json:"can_manage_inventory"`
	CanManageOrders    bool           `gorm:"default:true" json:"can_manage_orders"`
	CanChat            bool           `gorm:"default:true" json:"can_chat"`
	CanChangeStatus    bool           `gorm:"default:true" json:"can_change_status"`
	CanRate            bool           `gorm:"default:false" json:"can_rate"`
	StatusActive       bool           `gorm:"default:true" json:"status_active"`
	ProfilePicURL      string         `json:"profile_pic_url"`
	CreatedAt          time.Time      `json:"created_at"`
	UpdatedAt          time.Time      `json:"updated_at"`
	DeletedAt          gorm.DeletedAt `gorm:"index" json:"-"`
}

func (Employee) TableName() string {
	return "employees"
}
